ssh.go

```diff
@@ -69,16 +69,42 @@ type ServerConfigCallback func(ctx Context) *gossh.ServerConfig
 type ConnectionFailedCallback func(conn net.Conn, err error)
 
 // Window represents the size of a PTY window.
+// From https://datatracker.ietf.org/doc/html/rfc4254#section-6.2
+//
+// Zero dimension parameters MUST be ignored. The character/row dimensions
+// override the pixel dimensions (when nonzero).  Pixel dimensions refer
+// to the drawable area of the window.
 type Window struct {
-	Width  int
+	// Width in columns, overrides WidthPixels.
+	Width int
+	// Height in rows, overrides HeightPixels.
 	Height int
+
+	// These represent the drawable area of the window.
+	WidthPixels  int
+	HeightPixels int
 }
 
 // Pty represents a PTY request and configuration.
 type Pty struct {
-	Term   string
+	// Term is the TERM environment variable value.
+	Term string
+
+	// Windows sent as part of the pty-req.
 	Window Window
-	// HELP WANTED: terminal modes!
+
+	// Modes represent a mapping of Terminal Mode opcode to value as it was
+	// requested by the client as part of the pty-req. These are outlined as
+	// part of https://datatracker.ietf.org/doc/html/rfc4254#section-8.
+	//
+	// The keys/opcodes are defined as constants in golang.org/x/crypto/ssh (VINTR,VQUIT,etc.).
+	//
+	// The values are either boolean (0, 1), or uint32.
```

Comment:
Values are always uint32. Don't need to describe the type signature.

// Boolean opcodes have values 1 or 0.